### ISSUE TITLE
Gateway timeout (504) when trying to retrieve /api/v0/pins

### ISSUE THREAD

**Issue by giliomeejg** [ceramicnetwork/js-ceramic#2500]
# Description

On our production server, with around 60k streams, I'm getting a gateway timeout when POSTing to /api/v0/pins.  My intention is to get a list of streams, so that I can continue inspecting individual streams.

The lack of pagination when returning the result set is surely going to create problems down the line.

# Technical Information

Change /api/v0/pins, or any other endpoint that potentially returns long lists of data to return a paginated set, instead of the entire dataset.

**Comment by stbrody** [comment]
Yep, this is a known issue.  Listing the set of pinned streams returns the entire result set in a single HTTP response, which can be quite slow and unweildy when the pinset is large.  It's on our backlog to address, but isn't a high priority at the moment.
